### ISSUE TITLE
ImageDataGenerators don't produce the same filters.

### ISSUE THREAD

**Issue by faruknane** [keras-team/keras#12075]
Hi, I try to produce images with masks. There is a problem. It seems the seed parameter only affects shuffling? Because the code below doesn't produce the masks correctly. the masks and the images don't fit to each other. 

```
gc.collect()

data_gen_args = dict(rotation_range=40.,
width_shift_range=0.1,
height_shift_range=0.1,
zoom_range=0.2,
horizontal_flip=True,
fill_mode='nearest'
)

image_datagen = keras.preprocessing.image.ImageDataGenerator(**data_gen_args)

image_generator = image_datagen.flow(x_train[:,:,:,1:4],
                                     seed=1,
                                     batch_size=5,
                                     shuffle=True)


mask_datagen = keras.preprocessing.image.ImageDataGenerator(**data_gen_args)



mask_generator = mask_datagen.flow(y_train,
                                   seed=1,
                                   batch_size=5,
                                   shuffle=True)

train_generator = zip(image_generator, mask_generator)
```

To print images:

```
print("basladi")
print(np.max(image_generator[0][0]))

for i in range(0, 5):
    plt.imshow(image_generator[0][i], cmap = plt.get_cmap('gray'))
    plt.show()
    plt.imshow(mask_generator[0][i].reshape(sizeXY,sizeXY), cmap = plt.get_cmap('gray'))
    plt.show()
```

The problem is that the generators don't apply the same filters to the images and the masks.

**Comment by gabrieldemarmiesse** [comment]
@Dref360 Any insight on this?
